columnphysics/icepack_kinds.F90

```diff
@@ -17,10 +18,18 @@ module icepack_kinds
                             char_len_long  = 256, &
                             log_kind  = kind(.true.), &
                             int_kind  = selected_int_kind(6), &
+#if defined (NO_I8)
+                            int8_kind = selected_int_kind(6), &
+#else
                             int8_kind = selected_int_kind(13), &
+#endif
                             real_kind = selected_real_kind(6), &
                             dbl_kind  = selected_real_kind(13), &
+#if defined (NO_R16)
+                            r16_kind  = selected_real_kind(13)
+#else
                             r16_kind  = selected_real_kind(33,4931)
+#endif
```

Comment:
The NO_I8 and NO_R16 are definitely compiler dependent which means they may be required on certain machines.  And this could certainly change.  For instance, NO_R16 is defined for a few machine/compilers now.  It's possible those are incorrect.  To be honest, I'm not sure how those settings were decided.  I suspect it's partly historical and they have not been reviewed for a while.  The main point of these CPPs is that they provide users with a flag to turn off these features if their compiler doesn't support them.  As a general rule, they should only be used when needed.  And we probably should review the current settings in the current machines to check whether we really do need them where they are currently set.